File: 0392-is-subsequence/0392-is-subsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        idx1, idx2, n1, n2 = 0, 0, len(s), len(t)
        while idx1 < n1 and idx2 < n2:
            if s[idx1] == t[idx2]:
                idx1, idx2 = idx1 + 1, idx2 + 1
            else:
                idx2 += 1
        return idx1 is n1

File: 0392-is-subsequence/0392-is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int idx1 = 0, idx2 = 0, n1 = s.length(), n2 = t.length();
        while (idx1 < n1 && idx2 < n2) {
            if (s.charAt(idx1) == t.charAt(idx2)) {
                ++idx1; 
                ++idx2;
            } else 
                ++idx2;
        }
        return idx1 == n1;
    }
}